diff -Naur a/data/com.vixalien.muzika.appdata.xml.in.in b/data/com.vixalien.muzika.appdata.xml.in.in
--- a/data/com.vixalien.muzika.appdata.xml.in.in	2023-06-24 05:49:41.061270532 +0530
+++ b/data/com.vixalien.muzika.appdata.xml.in.in	2023-06-24 05:49:04.915876099 +0530
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop-application">
 	<id>@app-id@.desktop</id>
+	<icon type="stock">snap.muzika.@app-id@</icon>
 	<metadata_license>CC0-1.0</metadata_license>
 	<project_license>GPL-3.0+</project_license>
 	<name>Muzika</name>
@@ -9,7 +10,7 @@
 		<p>
 			Play music from millions of songs available on the internet for free.
 			Search for songs, albums, artists, and more. Discover new music from
-			playlists and get a personalized &amp; curated experience. 
+			playlists and get a personalized &amp; curated experience.
 		</p>
 		<p>Current features:</p>
 		<ul>
@@ -50,4 +51,4 @@
 	<content_rating type="oars-1.1" />
 	<provides>@app-id@.desktop</provides>
 	<translation type="gettext">@app-id@</translation>
-</component>
\ No newline at end of file
+</component>
diff -Naur a/data/com.vixalien.muzika.desktop.in b/data/com.vixalien.muzika.desktop.in
--- a/data/com.vixalien.muzika.desktop.in	2023-06-24 05:48:20.112064324 +0530
+++ b/data/com.vixalien.muzika.desktop.in	2023-06-08 02:33:30.736989265 +0530
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=Muzika
 Exec=com.vixalien.muzika
-Icon=com.vixalien.muzika
+Icon=snap.muzika.com.vixalien.muzika
 Terminal=false
 Type=Application
 Categories=GTK;
diff -Naur a/src/com.vixalien.muzika.in b/src/com.vixalien.muzika.in
--- a/src/com.vixalien.muzika.in	2023-06-08 02:30:18.732825202 +0530
+++ b/src/com.vixalien.muzika.in	2023-06-08 02:34:12.048854100 +0530
@@ -1,4 +1,4 @@
-#!@GJS@ -m
+#!/snap/muzika/current/gnome-platform/usr/bin/gjs -m
 
 import GLib from "gi://GLib";
 import { exit } from "system";
